jsHelper/spicetifyWrapper.js

```diff
@@ -1025,34 +1025,37 @@ Spicetify.SVGIcons = {
 };
 
 (async function waitUserAPI() {
-	if (!Spicetify.Platform?.UserAPI?._product_state) {
+	if (!Spicetify.Platform?.UserAPI) {
 		setTimeout(waitUserAPI, 1000);
 		return;
 	}
 
 	let subRequest;
 
+	// product_state was renamed to product_state_service in Spotify 1.2.21
+	const productState = Spicetify.Platform.UserAPI._product_state || Spicetify.Platform.UserAPI._product_state_service;
```

Comment:
actually, you should leave the optional chaining
```suggestion
	const productState = Spicetify.Platform.UserAPI?._product_state || Spicetify.Platform.UserAPI?._product_state_service;
```

Reply:
This line will not be executed if UserAPI is undefined, will it?

Reply:
Huh. JavaScript can simply throw error because it will try to access variable that doesn't exist in this object - it's better to be safe